Fluid/Ast/PaginateStatement.cs

```diff
@@ -0,0 +1,249 @@
+﻿using Fluid.Values;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+
+namespace Fluid.Ast
+{
+    public class PaginateStatement : Statement
+    {
+        private readonly Expression _expression;
+        private readonly long _pageSize;
+        private readonly List<Statement> _statements;
+
+        public PaginateStatement(Expression expression, long pageSize, List<Statement> statements)
+        {
+            _expression = expression ?? throw new ArgumentNullException(nameof(expression));
+            _pageSize = pageSize;
+            _statements = statements ?? new List<Statement>();
+        }
+
+        public long PageSize => _pageSize;
+
+        public override async ValueTask<Completion> WriteToAsync(TextWriter writer, TextEncoder encoder, TemplateContext context)
+        {
+            var value = await _expression.EvaluateAsync(context);
+            if (value == null || value is not PaginateableValue paginateableValue) return Completion.Normal;
```

Comment:
I think this should work with standard collection values (Array, Dictionary, even strings) but it should handle a custom `IPaginate` if necessary to customize the results (somehow like you did here).

Reply:
IPaginate needs to return the current page, but the standard collection does not provide this value (unless we agree on a value in the context). It also requires a method to generate a URL to populate the paginate object.